pkg/installer/spec.sh

```diff
@@ -0,0 +1,620 @@
+#!/bin/sh
```

Comment:
For now this is the same file of pkg/debug.... is there any particular reason for the copy? Or maybe you could import directly from the pkg/debug container to avoid duplication here, it's very easy to get out of sync when someone update the pkg/debug/spec.sh....

Same question for the pkg/isntaller/install from pkg/mkimage-raw-efi

Reply:
> Don't we remove the original files? I thought we did

Files that were in `pkg/mkimage-raw-efi` that moved over, and files that were in `pkg/mkverification-raw-efi`, we did remove the originals.

This one particularly comes from `pkg/debug/`, and needs to stay there.

> import directly from the pkg/debug container

How? We have no mechanism to import from one pkg to its peer. I spent quite a bit of time on this particular part, and found no easy solution. The original `install` script (in `pkg/mkimage-raw-efi/`) actually bind mounted the file from `/containers/..../debug/`, which was horrible and brittle.

I thought about making it part of a library and then having both consume it, but that turned out to be a bigger change, and this PR already is pretty big.

To the best of my knowledge, all of this should go away RSN (real soon now), so I was willing to live with the duplication.

Reply:
> > import directly from the pkg/debug container
> 
> How? We have no mechanism to import from one pkg to its peer. I spent quite a bit of time on this particular part, and found no easy solution. The original `install` script (in `pkg/mkimage-raw-efi/`) actually bind mounted the file from `/containers/..../debug/`, which was horrible and brittle.
> 

I was thinking about just take spec.sh from the debug container, like this...

```diff
diff --git a/pkg/installer/Dockerfile b/pkg/installer/Dockerfile
index 6efed25bc..10426876e 100644
--- a/pkg/installer/Dockerfile
+++ b/pkg/installer/Dockerfile
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Dockerfile to build installer img initrd
+FROM lfedge/eve-debug:87da9d836425558144381d204795fa4aee74efed AS debug
 FROM lfedge/eve-alpine:c114cf1d3ea51534f061f9aa949beb6ac5c12fb3 AS build
 SHELL ["/bin/ash", "-eo", "pipefail", "-c"]
 ENV BUILD_PKGS mkinitfs grep patch make coreutils musl-dev gcc g++ perl \
@@ -62,9 +63,10 @@ RUN ./bootstrap \
     /out/ddcutil-1.2.2 /out/hw-probe-1.6-AI /out/tpm2-tools-5.2
 
 FROM scratch
+COPY --from=debug /usr/bin/spec.sh /
 COPY --from=build /out/ /
 COPY install /
-COPY spec.sh verify verifytpm.sh /
+COPY verify verifytpm.sh /
 
 WORKDIR /
 RUN echo "mtools_skip_check=1" >> etc/mtools.conf
@@ -75,4 +77,4 @@ ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 # hadolint ignore=DL3002
 USER 0:0
 
-ENTRYPOINT [ "/install" ]
\ No newline at end of file
+ENTRYPOINT [ "/install" ]
```

But if you tell that all of this should be go away soon, that's fine, we can live with this duplication.....

Reply:
I don't mind that, but then it is hard-coded in here. Will we remember to update the installer container every time we update debug container? Right now, the only place we do that is in `images/*.yml`, where we have `parse-pkgs.sh` called from the Makefile. I was very hesitant of adding a whole new mechanism in here.

I recommend we leave the copy here. It all should go away soon. And if it doesn't, we can create a library pkg just for that, which both debug and installer can consume. But that would be an incredibly small PR compared to this. 😆